### PR TITLE
Add py-pcircle - a suite of parallel file system tools

### PR DESCRIPTION
**Pull request by fwang2**

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-pcircle/package.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+from spack import *
+
+
+class PyPcircle(PythonPackage):
+    """ A suite of parallel file system tools."""
+    homepage = "http://github.com/olcf/pcircle"
+    version('develop', git='http://github.com/olcf/pcircle.git',
+            branch='dev')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-pcircle/package.py]: It looks like an official release was just added. Can you add that to this package?

File: var/spack/repos/builtin/packages/py-pcircle/package.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+    import_modules = ['openmpi', 'python']
+
+    phases = ['install']
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-pcircle/package.py]: Was the 'build' phase causing problems?

File: var/spack/repos/builtin/packages/py-pcircle/package.py [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+    depends_on('python@2.7.14:')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-pcircle/package.py]: It looks like Python 3 doesn't work:
```python
if (3, 0, 0) < sys.version_info[:3] < (2, 7, 0):                                
    raise RuntimeError("This application requires Python 2.7.x")  
```
This line should read:
```python
depends_on('python@2.7:2.8')
```

File: var/spack/repos/builtin/packages/py-pcircle/package.py [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+    depends_on('py-mpi4py')
+    depends_on('py-scandir')
+    depends_on('py-xattr')
+    depends_on('py-numpy')
+    depends_on('py-bitarray')
+    depends_on('py-cffi', type='build')
+    depends_on('py-setuptools', type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-pcircle/package.py]: Missing a dependency on `py-lru-dict`. Can you also add the specific version requirements?
```python
requires = [                                                                    
    'cffi>=1.2.1',                                                              
    'mpi4py>=1.3.1',                                                            
    'lru-dict>=1.1.1',                                                          
    'xattr',                                                                    
    'scandir>=1.1',                                                             
    'numpy>=1.9.2',                                                             
    'bitarray>=0.8.1'                                                           
]             
```
Also, all of these (except `py-cffi`) should be `type=('build', 'run')` as they don't need to be linked.

**Comment by alalazo** [comment]
Closing as this has been stale for more than a year. I'm not marking this as `up-for-grabs` since the package seems to be not widespread + it had no activity in the last year. Feel free to reopen if you want to finish the PR.

**Pull request closed without merging**